src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -29,8 +29,37 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
     MTRDeviceStateUnreachable = 2
 };
 
+/**
+ * This enum is used to specify the type of logs requested from this device.
+ *
+ * The logs types are : End User Support, Network Diagnostics and Crash logs.
+ */
+typedef NS_ENUM(NSUInteger, MTRDiagnosticLogType) {
+    MTRDiagnosticLogTypeEndUserSupport = 0, // End user support logs are requested
+    MTRDiagnosticLogTypeNetworkDiagnostics = 1, // Network Diagnostics logs are requested
+    MTRDiagnosticLogTypeCrash = 2, // Crash logs are requested
+    MTRDiagnosticLogTypeUnknown = 3
+} MTR_PROVISIONALLY_AVAILABLE;
+
 @protocol MTRDeviceDelegate;
 
+MTR_PROVISIONALLY_AVAILABLE
+@interface MTRDiagnosticLogResult : NSObject
+
+/**
+ * A boolean indicating whether the request for diagnostic logs has timed out.
+ */
+@property (readonly, nonatomic) BOOL requestTimedOut;
+
+/**
+ * Array containing the file path(s) for the desired log type(s).
+ * The log type values should correspond to the enum MTRDiagnosticLogType values.
+ * If there are no logs an empty array will be returned.
+ */
+@property (readonly, nonatomic) NSDictionary<NSNumber *, NSString *> * filePathDictionary;
```

Comment:
As discussed on Slack, I think having separate properties might be nicer, i.e.
```
@property (readonly, nullable, nonatomic) NSURL *endUserSupportLog;
@property (readonly, nullable, nonatomic) NSURL *networkDiagnosticsLog;
@property (readonly, nullable, nonatomic) NSURL *crashLog;
```

If sticking with the dictionary, I think it should just be called something like `files` (without the word "dictionary")

Reply:
(you could use a dictionary internally if that's simpler to implement and each of those getters would be a one-liner)